istioctl/cmd/workload.go

```diff
@@ -0,0 +1,399 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	clientv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	networkingclientv1alpha3 "istio.io/client-go/pkg/clientset/versioned/typed/networking/v1alpha3"
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/inject"
+	"istio.io/istio/pkg/util/gogoprotomarshal"
+
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	tokenDuration  int64
+	name           string
+	serviceAccount string
+	filename       string
+	outputName     string
+	clusterID      string
+	ports          []string
+)
+
+const (
+	tempPerms = os.FileMode(0644)
+)
+
+func workloadCommands() *cobra.Command {
+	workloadCmd := &cobra.Command{
+		Use:   "workload",
+		Short: "Commands to assist in configuring and deploying workloads",
+	}
+	workloadCmd.AddCommand(groupCommand())
+	workloadCmd.AddCommand(entryCommand())
+	return workloadCmd
+}
+
+func groupCommand() *cobra.Command {
+	groupCmd := &cobra.Command{
+		Use:   "group",
+		Short: "Commands dealing with workload groups",
+	}
+	groupCmd.AddCommand(createCommand())
+	return groupCmd
+}
+
+func entryCommand() *cobra.Command {
+	entryCmd := &cobra.Command{
+		Use:   "entry",
+		Short: "Commands dealing with workload entries",
+	}
+	entryCmd.AddCommand(configureCommand())
+	return entryCmd
+}
+
+func createCommand() *cobra.Command {
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a WorkloadGroup YAML artifact representing workload instances",
```

Comment:
`Creates a WorkloadGroup resource that provides a template for associated WorkloadEntries`?